utils.js

```diff
@@ -59,43 +59,45 @@ export function convertMillisecondsToFormattedTime(milliseconds) {
     return formattedTime;
 }
 
-const BASE62_CHARS = '0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz';
+const BASE64_CHARS = '0123456789abcdef-ghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ_';
 
 /**
- * 
- * @param {string} num 
- * @param {number} [base=10] Starting base of number. Max is Base-32. Any value higher will come out as Base-32
- * @returns {string}
+ * Function that converts a number from up to Base-64 to a decimal
+ * @param {string} str
+ * @param {string} base Base in which you're converting from
+ * @returns {BigInt}
  */
-export function convertNumberToBase62(num, base = 10) {
-    base = parseInt(Math.min(Math.max(base, 2), 32), 10);
-    let decimalVal = parseInt(num, base);
-    let base62Val = '';
-
-    while (decimalVal > 0) {
-        const remainder = decimalVal % 62;
-        base62Val = BASE62_CHARS.charAt(remainder) + base62Val;
-        decimalVal = Math.floor(decimalVal / 62);
+function convertToDecimal(str, base) {
+    let res = 0n;
+
+    for (let i = 0; i < str.length; i++) {
+        const char = str[str.length - 1 - i];
+        const digitVal = BASE64_CHARS.indexOf(char);
+
+        res += BigInt(digitVal) * (BigInt(base) ** BigInt(i));
```

Comment:
please for the love of god add some unit tests to this code of doom or i might cry. it can just be the stuff you were doing earlier but in a test file

https://jestjs.io/docs/getting-started

Reply:
i did it